pkg/csi/service/common/util.go

```diff
@@ -418,14 +417,17 @@ func GetClusterComputeResourceMoIds(ctx context.Context) ([]string, error) {
 		return nil, fmt.Errorf("could not find any AvailabilityZone")
 	}
 
+	var (
+		clusterComputeResourceMoIds []string
+		found                       bool
+	)
 	for _, az := range azList.Items {
-		// TODO: TKGS-HA - convert to slice when appropriate
-		clusterComputeResourceMoId, found, err := unstructured.NestedString(az.Object, "spec", "clusterComputeResourceMoId")
+		clusterComputeResourceMoIds, found, err = unstructured.NestedStringSlice(az.Object, "spec",
+			"clusterComputeResourceMoIDs")
 		if !found || err != nil {
```

Comment:
same [comment](https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1689/files#r845433486) applies here as well.

Reply:
done

Reply:
but if we return an empty slice with a nil error, are we handling the case for all callers?